packages/generator/src/generator-without-ts-morph/request-builder/class.ts

```diff
@@ -0,0 +1,247 @@
+import { codeBlock, documentationBlock, unixEOL } from '@sap-cloud-sdk/util';
+import { getFunctionDoc, getRequestBuilderDescription } from '../../typedoc';
+import { VdmEntity } from '../../vdm-types';
+
+/**
+ * @internal
+ */
+export function requestBuilderClass(entity: VdmEntity): string {
+  return (
+    unixEOL +
+    documentationBlock`${getRequestBuilderDescription(entity)}` +
+    unixEOL +
+    codeBlock`export class ${
+      entity.className
+    }RequestBuilder<T extends DeSerializers = DefaultDeSerializers> extends RequestBuilder<${
+      entity.className
+    }<T>, T>{
+
+    ${requestBuilderMethods(entity).join('\n\n')}
+
+  }`
+  );
+}
+
+function requestBuilderMethods(entity: VdmEntity): string[] {
+  const methods = [
+    getByKeyRequestBuilder(entity),
+    getAllRequestBuilder(entity)
+  ];
+  if (entity.creatable) {
+    methods.push(createRequestBuilder(entity));
+  }
+
+  if (entity.updatable) {
+    methods.push(updateRequestBuilder(entity));
+  }
+
+  if (entity.deletable) {
+    methods.push(deleteRequestBuilder(entity));
+  }
+  return methods;
+}
+
+function getByKeyRequestBuilder(entity: VdmEntity): string {
+  return (
+    documentationBlock`${getFunctionDoc(
+      `Returns a request builder for retrieving one \`${entity.className}\` entity based on its keys.`,
+      {
+        returns: {
+          type: `GetByKeyRequestBuilder<${entity.className}>`,
+          description: `A request builder for creating requests to retrieve one \`${entity.className}\` entity based on its keys.`
+        },
+        params: entity.keys.map(key => ({
+          name: key.propertyNameAsParam,
+          type: key.jsType,
+          description: `Key property. See {@link ${entity.className}.${key.instancePropertyName}}.`
+        }))
+      }
+    )}` +
+    unixEOL +
+    codeBlock`getByKey(${entity.keys
+      .map(
+        key =>
+          `${key.propertyNameAsParam}: DeserializedType<T, '${key.edmType}'>`
+      )
+      .join(', ')}): GetByKeyRequestBuilder<${entity.className}<T>, T> {
+        ${buildParametrizedStatements(entity, 'GetByKeyRequestBuilder')}
+  }`
+  );
+}
+
+function getAllRequestBuilder(entity: VdmEntity): string {
+  return (
+    documentationBlock`${getFunctionDoc(
+      `Returns a request builder for querying all \`${entity.className}\` entities.`,
+      {
+        returns: {
+          type: `GetAllRequestBuilder<${entity.className}>`,
+          description: `A request builder for creating requests to retrieve all \`${entity.className}\` entities.`
+        }
+      }
+    )}` +
+    unixEOL +
+    codeBlock`getAll(): GetAllRequestBuilder<${entity.className}<T>, T> {
+    return new GetAllRequestBuilder<${entity.className}<T>, T>(this.entityApi);
+  }`
+  );
+}
+
+function createRequestBuilder(entity: VdmEntity): string {
```

Comment:
[pp] you can deconstruct the className from the entity in the argument or directly only pass the class name in. Also applies to other methods.

Reply:
Perhaps destructuring in the argument is even nicer, because you have a short `className` var name but you see where it came from in the method declaration.